File: backend/app/services/trn_service.py
import re

TRN_PATTERN = re.compile(r"^\d{9}$")


def normalize_trn(value: str | None) -> str | None:
    if value is None:
        return None
    raw = str(value).strip()
    if not raw:
        return None
    digits = re.sub(r"\D", "", raw)
    if not TRN_PATTERN.match(digits):
        raise ValueError("TRN must be 9 digits")
    return digits


def format_trn(trn: str) -> str:
    n = normalize_trn(trn)
    if not n:
        raise ValueError("TRN must be 9 digits")
    return f"{n[:3]}-{n[3:6]}-{n[6:]}"

File: server/src/services/trnService.ts
const TRN_PATTERN = /^\d{9}$/

export function normalizeTrn(value: string | null | undefined): string | null {
  if (value == null) return null
  const raw = String(value).trim()
  if (!raw) return null
  const digits = raw.replace(/\D/g, '')
  if (!TRN_PATTERN.test(digits)) {
    throw new Error('TRN must be 9 digits')
  }
  return digits
}

export function formatTrn(trn: string): string {
  const n = normalizeTrn(trn)
  if (!n) throw new Error('TRN must be 9 digits')
  return `${n.slice(0, 3)}-${n.slice(3, 6)}-${n.slice(6)}`
}
